feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -136,6 +159,66 @@ class PreviewViewModel @AssistedInject constructor(
         }
     }
 
+    /**
+     * Returns the difference between two [CameraAppSettings] as a mapping of <[KProperty], [Any]>.
+     */
+    private fun getSettingsDiff(
+        oldCameraAppSettings: CameraAppSettings,
+        newCameraAppSettings: CameraAppSettings
+    ): Map<KProperty<Any?>, Any?> {
+        val diffSettingsMap = HashMap<KProperty<Any?>, Any?>()
+        CameraAppSettings::class.memberProperties.forEach { property ->
+            if (property.get(oldCameraAppSettings) != property.get(newCameraAppSettings)) {
+                diffSettingsMap[property] = property.get(newCameraAppSettings)
+            }
+        }
+        return diffSettingsMap
+    }
+
+    /**
+     * Iterates through a queue of [Pair]<[KProperty], [Any]> and attempt to apply them to
+     * [CameraUseCase].
+     */
+    private suspend fun applySettingsDiff(diffSettingsMap: Map<KProperty<Any?>, Any?>) {
+        diffSettingsMap.keys.forEach { key ->
+            when (key) {
+                CameraAppSettings::cameraLensFacing -> {
+                    cameraUseCase.setLensFacing(diffSettingsMap[key] as LensFacing)
+                }
+
+                CameraAppSettings::flashMode -> {
+                    cameraUseCase.setFlashMode(diffSettingsMap[key] as FlashMode)
+                }
+
+                CameraAppSettings::captureMode -> {
+                    cameraUseCase.setCaptureMode(diffSettingsMap[key] as CaptureMode)
+                }
+
+                CameraAppSettings::aspectRatio -> {
+                    cameraUseCase.setAspectRatio(diffSettingsMap[key] as AspectRatio)
+                }
+
+                CameraAppSettings::previewStabilization -> {
+                    cameraUseCase.setPreviewStabilization(diffSettingsMap[key] as Stabilization)
+                }
+
+                CameraAppSettings::videoCaptureStabilization -> {
+                    cameraUseCase.setVideoCaptureStabilization(
+                        diffSettingsMap[key] as Stabilization
+                    )
+                }
+
+                CameraAppSettings::targetFrameRate -> {
+                    cameraUseCase.setTargetFrameRate(diffSettingsMap[key] as Int)
+                }
+
+                CameraAppSettings::darkMode -> {}
+
+                else -> TODO("Unhandled CameraAppSetting $key")
+            }
+        }
+    }
+
```

Comment:
nit: it's generally better to operate on entries rather than using keys and looking up from the map. This ensures that entries don't change (or get removed) between getting the key and looking up in the map:
```
   private suspend fun applySettingsDiff(diffSettingsMap: Map<KProperty<Any?>, Any?>) {
        diffSettingsMap.entries.forEach { entry ->
            when (entry.key) {
                CameraAppSettings::cameraLensFacing -> {
                    cameraUseCase.setLensFacing(entry.value as LensFacing)
                }

                CameraAppSettings::flashMode -> {
                    cameraUseCase.setFlashMode(entry.value as FlashMode)
                }

                CameraAppSettings::captureMode -> {
                    cameraUseCase.setCaptureMode(entry.value as CaptureMode)
                }

                CameraAppSettings::aspectRatio -> {
                    cameraUseCase.setAspectRatio(entry.value as AspectRatio)
                }

                CameraAppSettings::previewStabilization -> {
                    cameraUseCase.setPreviewStabilization(entry.value as Stabilization)
                }

                CameraAppSettings::videoCaptureStabilization -> {
                    cameraUseCase.setVideoCaptureStabilization(entry.value as Stabilization)
                }

                CameraAppSettings::targetFrameRate -> {
                    cameraUseCase.setTargetFrameRate(entry.value as Int)
                }

                CameraAppSettings::darkMode -> {}

                else -> TODO("Unhandled CameraAppSetting $entry")
            }
        }
    }
```